kernelci/dashboard
# feat: consider env_misc.platform as hardware

Add a new logic to the backend where `environment_misc.platform` will work as a fallback case for when the `environment_compatabile` field is null in the database. When `environment_misc.plaftorm` is null, we'll use `build.misc.platform`, but the logic of the Hardware filter will be kept the same, meaning that the builds returned will be first the build associated with the tests that have that hardware. This means that even if `build.misc.platform` is null but the build is associated with a test that have `environment_compatible` or `environment_misc.platform` not null, it will also be added when filtering for that hardware

This drastically lowers the number of Unknown hardware but they can still occur if both `environment_compatible` and `environment_misc.platform` and `build.misc.platform` are null.

Closes #692 

## How to test
- Add hardware filters from the Hardware Used card, Hardware Tested card and Filter modal and verify that they are correctly being added by comparing with staging. Verify that the commit graph is also working properly, showing the correct graph and changing the commit correctly
- Add a platform filter (these are hardwares that do not appear in staging, e.g. `acer-R721T-grunt`) from the Hardware Used card, Hardware Tested card and Filter modal and make the same verifications as above, if the status are showing correctly (comparing the values in the Hardware Tested card and the value shown in the pie chart) and if the commit graph is working.
- Test in other origins to verify that the behaviour is consistent between them
